partiql-ast/src/main/java/org/partiql/ast/literal/Literal.java

```diff
@@ -0,0 +1,125 @@
+package org.partiql.ast.literal;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.ast.AstNode;
+import org.partiql.ast.AstVisitor;
+import org.partiql.ast.DataType;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Collection;
+import java.util.Collections;
+
+public abstract class Literal extends AstNode {
+    @NotNull
+    public abstract LiteralKind kind();
+
+    @Override
+    @NotNull
+    public Collection<AstNode> children() {
+        return Collections.emptyList();
+    }
```

Comment:
In a quick follow-up PR, could you please change this to

```
public List<AstNode> getChildren() {
    ...
}
```

This way it has ordered semantics and the `getFoo` while clunky (imo) plays better with kotlin accessors.

Reply:
Sure. I'll track this as a followup in https://github.com/partiql/partiql-lang-kotlin/issues/1610.